### ISSUE TITLE
Expose analysis layers in viz.json v3

### ISSUE THREAD

**Issue by juanignaciosl** [CartoDB/cartodb#6846]
Related to #6751.
- [X] Expose analyses.
- [X] Do not expose analyses at vizjson for named maps (they will be at the template).
- [ ] ~~Layers with analyses have source instead of sql.~~ #7061

Proposed structure both for layergroups and vizjson.

```
{
  layers: [ … ],
  dataviews: [ … ], // vizjson contains widgets, layergroup contains dataviews
  analyses: [ … ]
}
```

**Comment by alonsogarciapablo** [comment]
From the POV of CartoDB.js, exposing an `analyses` key at the top level for anonymous maps should be sufficient. Something along the lines of:

```
{
  id: 'something',
  // ... 
  layers: [
    { // ... }
  ],
  analyses: [{
    id: 'a1',
    type: 'point-in-polygon',
    params: {
      points_source: {
        id: 'a0',
        type: 'source',
        params: {
          query: 'SELECT * FROM cities'
        }
      },
      polygons_source: {
        id: 'a2',
        type: 'source',
        params: {
          query: 'SELECT * FROM states'
        }
      }
    }
  ]}
}
```

**NOTE: Analyses should only be present in the case of anonymous maps**. Named maps should already have all the analyses related stuff in the template.

**Comment by alonsogarciapablo** [comment]
@juanignaciosl I think there are more changes that need to be done. For example, layers that have analyses will need to point to them as their source (instead of the existing `sql` attribute). I think it makes sense to make everything work on the CartoDB.js side before making this changes in the viz.json. I will keep you posted.

**Comment by juanignaciosl** [comment]
Ok, will wait, then. Thanks!

**Comment by juanignaciosl** [comment]
Closed via #7034.
